woocommerce/woocommerce-ios
# Dynamic Dashboard: Fix layout for chart x-axis

Closes: #12531 


## Description

Previously we set a fixed value of 5 for strides of labels on the x-axis of the store stats chart. This causes an issue where the stride is too large for a custom range, causing only one label to be present on the x-axis.

This PR fixes the above issue by calculating the stride for x-axis values based on the length and granularity of each custom range. The layout of the chart has been updated to avoid any labels being clipped off.

Also: fixed the color of the navigation bar as reported in p1713856652563119-slack-C03L1NF1EA3.

## Testing instructions

- Enable the `dynamicDashboard` feature flag and build the app.
- Log in to a store with revenues and enable the Performance card on the dashboard screen.
- Add a custom range for the Performance card and confirm that the chart looks correct with x-axis labels.
- Try switching different range to confirm the x-axis looks correct for every length and granularity.

Testing navigation bar fix:
- Scroll the Dashboard screen up on both dark and light mode to confirm that the navigation bar color is correct.

## Screenshots



Before | After
--- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/139eb280-fdd3-4357-baa2-bcfbdfa81d76" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/ed7d1f06-3876-4623-8d02-c8924aba7347" width=320 />


---
